receiver/awscontainerinsightreceiver/internal/host/ebsvolume.go

```diff
@@ -0,0 +1,248 @@
+				ebsVolume.refresh()
+			case <-ebsVolume.shutdownC:
+				return
+			}
+		}
+	}()
+
+	return ebsVolume
+}
+
+func (e *EbsVolume) Shutdown() {
+	close(e.shutdownC)
+}
+
+func (e *EbsVolume) refresh() {
+	if e.instanceID == "" {
+		return
+	}
+
+	e.logger.Info("Fetch ebs volumes from ec2 api")
+	sess, err := session.NewSession(&aws.Config{})
+	if err != nil {
+		e.logger.Warn("Fail to set up session to call ec2 api", zap.Error(err))
+	}
+	client := ec2.New(sess)
+
+	input := &ec2.DescribeVolumesInput{
+		Filters: []*ec2.Filter{
+			{
+				Name:   aws.String("attachment.instance-id"),
+				Values: aws.StringSlice([]string{e.instanceID}),
+			},
+		},
+	}
+
+	devPathSet := make(map[string]bool)
+	allSuccess := false
+	for {
+		result, err := client.DescribeVolumes(input)
+		if err != nil {
+			e.logger.Warn("Fail to call ec2 DescribeVolumes", zap.Error(err))
+		}
+		for _, volume := range result.Volumes {
+			for _, attachment := range volume.Attachments {
+				devPath := e.addEbsVolumeMapping(volume.AvailabilityZone, attachment)
+				devPathSet[devPath] = true
+			}
+		}
+		allSuccess = true
+		if result.NextToken == nil {
+			break
+		}
+		input.SetNextToken(*result.NextToken)
+	}
+
+	if allSuccess {
+		e.Lock()
+		defer e.Unlock()
+		for k := range e.dev2Vol {
+			if !devPathSet[k] {
+				delete(e.dev2Vol, k)
+			}
+		}
+	}
+}
+
+func (e *EbsVolume) addEbsVolumeMapping(zone *string, attachement *ec2.VolumeAttachment) string {
+	// *attachement.Device is sth like: /dev/xvda
+	devPath := findNvmeBlockNameIfPresent(*attachement.Device)
+	if devPath == "" {
+		devPath = *attachement.Device
+	}
+
+	e.Lock()
+	defer e.Unlock()
+	e.dev2Vol[devPath] = fmt.Sprintf("aws://%s/%s", *zone, *attachement.VolumeId)
+	return devPath
+}
+
+// find nvme block name by symlink, if symlink doesn't exist, return ""
+func findNvmeBlockNameIfPresent(devName string) string {
+	// for nvme(ssd), there is a symlink from devName to nvme block name, i.e. /dev/xvda -> /dev/nvme0n1
+	// https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/nvme-ebs-volumes.html
+	hasRootFs := true
+	if _, err := os.Lstat("/rootfs/proc"); os.IsNotExist(err) {
+		hasRootFs = false
+	}
+	nvmeName := ""
+
+	if hasRootFs {
+		devName = "/rootfs" + devName
+	}
+
+	if info, err := os.Lstat(devName); err == nil {
+		if info.Mode()&os.ModeSymlink != 0 {
+			if path, err := filepath.EvalSymlinks(devName); err == nil {
+				nvmeName = path
+			}
+		}
+	}
+
+	if nvmeName != "" && hasRootFs {
+		nvmeName = strings.TrimPrefix(nvmeName, "/rootfs")
+	}
+	return nvmeName
+}
+
+func (e *EbsVolume) GetEbsVolumeID(devName string) string {
+	e.RLock()
+	defer e.RUnlock()
+
+	for k, v := range e.dev2Vol {
+		// The key of dev2Vol is device name like nvme0n1, while the input devName could be a partition name like nvme0n1p1
+		if strings.HasPrefix(devName, k) {
+			return v
+		}
+	}
+
+	return ""
+}
+
+//extract the ebs volume id used by kubernetes cluster
+func ExtractEbsIDsUsedByKubernetes() map[string]string {
```

Comment:
Do you want to return error here? Missing mount means the user didn't set the right mount in their manifest (ecs task def or k8s pod). Thought it is possible user just don't want this metrics and skip the mount, in that case we might have some feature flag in the top level container insight config to switch things on/off.